androidApp/src/main/java/com/example/tasky/android/alarm/data/AlarmScheduler.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.tasky.android.alarm.data
+
+import android.app.AlarmManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import com.example.tasky.agenda.domain.model.AgendaItem
+import com.example.tasky.android.alarm.domain.IAlarmScheduler
+import kotlin.time.DurationUnit
+
+class AlarmScheduler(
+    private val context: Context,
+) : IAlarmScheduler {
+    private val alarmManager = context.getSystemService(AlarmManager::class.java)
+
+    override fun schedule(agendaItem: AgendaItem) {
+        val intent =
+            Intent(context, AlarmReceiver::class.java).apply {
+                putExtra(AlarmIntentKey.ID.name, agendaItem.id)
+                putExtra(AlarmIntentKey.TITLE.name, agendaItem.title)
+                putExtra(AlarmIntentKey.TYPE.name, agendaItem.getAgendaType())
+            }
+
+        alarmManager.setExact(
```

Comment:
With `setExact`, when the device is in low-power mode or in an idle state, then the system may defer the triggering of this alarm. For Tasky, this could cause the user to miss an event 😅 so this is not the behaviour we want. What we’d rather want to use is: `setExactAndAllowWhileIdle`. It is the most reliable function available to us to trigger an alarm at a specific time, regardless of the device’s current state (unless, of course, the device is off 😂 )